Simbody/src/CableSpan.cpp

```diff
@@ -0,0 +1,3028 @@
+    int constraintProjectionMaxIterations = 50;
+};
+
+//------------------------------------------------------------------------------
+//  Cable Span Parameters
+//------------------------------------------------------------------------------
+/* These are all the parameters for configuring the solver steps in the
+CableSpan. */
+struct CableSpanParameters final : IntegratorTolerances {
+    // The path's smoothness tolerance is defined as the angular misalignment
+    // at the points where the straight- and curved-line segments meet,
+    // measured in radians. When computing the optimal path this quantity is
+    // minimized, and the solver stops when reaching this tolerance.
+    Real smoothnessTolerance = 0.1 / 180. * Pi; // Default = 0.1 degrees.
+    // The solver's max allowed number of iterations used to compute the
+    // optimal path.
+    int solverMaxIterations = 50; // TODO a bit high?
+    // The solver's max allowed stepsize measured in radians. This is converted
+    // to a max allowed linear stepsize using the local radius of curvature
+    // evaluated at each obstacle.
+    Real solverMaxStepSize = 10. / 180. * Pi;
+};
+
+} // namespace
+
+//==============================================================================
+//                Contact Geometry Related Helper Functions
+//==============================================================================
+/* This section contains some helper functions for handling FrenetFrames and
+flipping some signs in ContactGeometry. */
+namespace
+{
+
+// Helper for grabbing the tangent axis.
+const UnitVec3& getTangent(const FrenetFrame& X)
+{
+    return X.R().getAxisUnitVec(TangentAxis);
+}
+
+// Helper for grabbing the normal axis.
+const UnitVec3& getNormal(const FrenetFrame& X)
+{
+    return X.R().getAxisUnitVec(NormalAxis);
+}
+
+// Helper for grabbing the binormal axis.
+const UnitVec3& getBinormal(const FrenetFrame& X)
+{
+    return X.R().getAxisUnitVec(BinormalAxis);
+}
+
+// Compute the frenet frame at a knot point of a geodesic.
+FrenetFrame calcFrenetFrameFromGeodesicState(
+    const ContactGeometry& geometry,
+    const ContactGeometry::GeodesicKnotPoint& q)
+{
+    FrenetFrame X;
+    X.updP() = q.point;
+    X.updR().setRotationFromTwoAxes(
+        geometry.calcSurfaceUnitNormal(q.point),
+        NormalAxis,
+        q.tangent,
+        TangentAxis);
+    return X;
+}
+
+// Compute the surface curvature such that tangentDot = curvature * normal
+// (This flips the sign).
+Real calcSurfaceCurvature(
+    const ContactGeometry& geometry,
+    const FrenetFrame& X_S,
+    CoordinateAxis direction)
+{
+    return -geometry.calcSurfaceCurvatureInDirection(
+        X_S.p(),
+        X_S.R().getAxisUnitVec(direction));
+}
+
+// Evaluate whether a point lies above the given surface.
+bool isPointAboveSurface(const ContactGeometry& geometry, const Vec3& point_S)
+{
+    // Try block was added here because some surfaces throw an exception when
```

Comment:
Probably should look into this, rather than pokemon-catching the exception and sucking it up. Because you can never be 100 % sure that the exception you're swalloing is from your code (as opposed to, say, an exception in a numeric library or additional implementation, OS, etc.)